pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java

```diff
@@ -94,6 +101,30 @@ public List<SegmentZKMetadata> getSegmentsZKMetadata(String tableNameWithType) {
     return ZKMetadataProvider.getSegmentsZKMetadata(_pinotHelixResourceManager.getPropertyStore(), tableNameWithType);
   }
 
+  /**
+   * Get all server to segment mappings for the given table.
+   *
+   * @param tableNameWithType Table name with type suffix
+   * @return Map where the key is the server and the value is a List of segments
+   */
+  public Map<String, List<String>> getServerToSegmentsMap(String tableNameWithType) {
```

Comment:
(minor) No need to add this API since we already have a getter for `_pinotHelixResourceManager`